docs/source/user-guide/example-usage.md

```diff
@@ -118,8 +118,8 @@ async fn main() -> datafusion::error::Result<()> {
   let ctx = SessionContext::new();
   let df = ctx.read_csv("tests/data/capitalized_example.csv", CsvReadOptions::new()).await?;
 
-  let df = df.filter(col("A").lt_eq(col("c")))?
-           .aggregate(vec![col("A")], vec![min(col("b"))])?
+  let df = df.filter(col("\"A\"").lt_eq(col("c")))?
+           .aggregate(vec![col("\"A\"")], vec![min(col("b"))])?
            .limit(0, Some(100))?;
```

Comment:
This is probably a pretty major change for people (that `col()` will now parse / normalize the column names according to SQL semantics).

However, it does have the nice side benefit that the handling of `.` is no longer treated differently in `col()`

What do you think about possibly 
1. changing `col` so that it forms a unqualified reference always (aka does not respect `.`)
2. adding some new function like `ident()` or `parse_col()` that treats column like a sql identifier and correctly splits based on `.` etc?

Reply:
Yeah that would have the benefit of being less breaking and being a bit more explicit in what is happening. `parse_col` would make it more explicit though I'm not really a fan of that name, feels a bit clunky.

For what its worth Spark does seem to be able to parse the input to their `col` function, instead of taking it wholly unqualified I believe

Reply:
How about we simply add an `ident()` function that doesn't parse at all? I can help do that

Reply:
I've added the `ident()` expr function and updated the `user-guide/example-usage.md` doc

Let me know if there are other places I need to change/add docs for